BackofficeBundle/EventListener/UpdateSiteNodesThemeListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace OpenOrchestra\BackofficeBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\PostFlushEventArgs;
+use Doctrine\ODM\MongoDB\Event\PreUpdateEventArgs;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\Model\SiteInterface;
+use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
+
+/**
+ * Class UpdateSiteNodesThemeListener
+ */
+class UpdateSiteNodesThemeListener
+{
+    protected $nodeClass;
+
+    protected $nodes = array();
+
+    /**
+     * @param string $nodeClass
+     */
+    public function __construct($nodeClass)
+    {
+        $this->nodeClass = $nodeClass;
+    }
+
+    /**
+     * @param PreUpdateEventArgs $event
+     */
+    public function preUpdate(PreUpdateEventArgs $event)
```

Comment:
you can't use this event in cms bundle, because with this event you have a dependence to MongoOdm